src/tools/server/images/local-icons-repository.ts

```diff
@@ -22,24 +27,30 @@ export class LocalIconsRepository extends AbstractIconRepository {
       };
     }
 
-    const files = fs.readdirSync('./public/icons');
+    const files = fs.readdirSync(iconsDirectory);
+    Consola.info(`Local icons repository directory exists and contains ${files.length} icons`);
 
     const normalizedEntries = files
       .filter((file) => ['.png', '.svg', '.jpeg', '.jpg'].some((x) => file.endsWith(x)))
       .map(
-        (file): NormalizedIcon => ({
-          name: file,
-          url: `./icons/${file}`,
-          size: 0,
-        })
+        (file): NormalizedIcon => {
+          const stats = fs.statSync(`${iconsDirectory}/${file}`);
+          return {
+            name: file,
+            url: `/icons/${file}`,
+            size: stats.size,
+          };
+        }
       );
 
     return {
       entries: normalizedEntries,
       count: normalizedEntries.length,
       success: true,
-      name: 'Local',
+      name: this.name,
       copyright: this.copyright,
     };
   }
+
+  protected name: string = "Local";
```

Comment:
Instead maybe pass through constructor

Reply:
Thanks for the suggestion. I'd like to have it parameter-less so I don't need a factory.